cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/ElasticsearchGenerator.scala

```diff
@@ -0,0 +1,430 @@
+          multiSearchQuery += ("joinSelectField" -> JsString(body.selectValues.head.name))
+          multiSearchQuery += ("joinTermsFilter" -> Json.toJson(joinTermsFilter))
+
+          return (ParsedResult(Seq.empty, exprMap), multiSearchQuery)
+        }
+        shallowQueryAfterAppend += ("groupAsList" -> groupAsArray)
+        (ParsedResult(Seq.empty, exprMap), shallowQueryAfterAppend)
+
+      case None =>
+        (ParsedResult(Seq.empty, exprMap), shallowQueryAfterAppend)
+    }
+  }
+
+  private def parseSelect(selectOpt: Option[SelectStatement],
+                          groupOpt: Option[GroupStatement],
+                          exprMap: Map[String, FieldExpr],
+                          queryAfterGroup: JsObject): (ParsedResult, JsObject) = {
+    var shallowQueryAfterGroup = queryAfterGroup
+
+    if (groupOpt.isEmpty && selectOpt.nonEmpty) {
+      val select = selectOpt.get
+      val orderStrs = select.orderOn.zip(select.order).map {
+        case (orderOn, order) =>
+          val expr = orderOn.name
+          val orderStr = if (order == SortOrder.DSC) "desc" else "asc"
+          s"""{"$expr": {"order": "$orderStr"}}"""
+      }
+      val orderStr =
+        if (orderStrs.nonEmpty) {
+          orderStrs.mkString("[", ",", "]")
+        } else {
+          ""
+        }
+      shallowQueryAfterGroup += ("size" -> JsNumber(select.limit))
+      shallowQueryAfterGroup += ("from" -> JsNumber(select.offset))
+      val source = select.fields.map(f => JsString(f.name))
+
+      if (orderStr.nonEmpty)
+        shallowQueryAfterGroup += ("sort" -> Json.parse(orderStr))
+      if (source.nonEmpty)
+        shallowQueryAfterGroup += ("_source" -> JsArray(source))
+    }
+    (ParsedResult(Seq.empty, exprMap), shallowQueryAfterGroup)
+  }
+
+  private def parseGlobalAggr(globalAggrOpt: Option[GlobalAggregateStatement],
+                              exprMap: Map[String, FieldExpr], queryAfterSelect: JsObject): JsObject = {
+    var shallowQueryAfterSelect = queryAfterSelect
+    globalAggrOpt match {
+      case Some(globalAggr) =>
+        val aggr = globalAggr.aggregate
+        val field = aggr.field.name
+        val as = aggr.as.name
+        val funcName = typeImpl.getAggregateStr(aggr.func)
+        val aggregatedJson = Json.parse(s"""{"func": "$funcName", "as": "$as"}""")
+        globalAggr.aggregate.func match {
+          case Count => {
+            shallowQueryAfterSelect += {"aggregation" -> aggregatedJson}
+            if (field != "*") {
+              shallowQueryAfterSelect += ("_source" -> Json.arr(JsString(field)))
+            }
+          }
+          case Min | Max => {
+            shallowQueryAfterSelect += ("aggregation" -> aggregatedJson)
+            shallowQueryAfterSelect += ("size" -> JsNumber(0))
+            shallowQueryAfterSelect += ("aggs" -> Json.obj( as -> Json.obj(funcName -> Json.obj("field" -> JsString(field)))))
+          }
+        }
+      case None => return shallowQueryAfterSelect
```

Comment:
Same here: what about `SUM`, `AVG`?

Reply:
If we'll not implement `SUM`, `AVG` in this PR, let's add one todo comment here.

Reply:
Ok, I added TODO to our code